zkevm-circuits/src/anchor_tx_circuit/sign_verify.rs

```diff
@@ -0,0 +1,424 @@
+    pub(crate) fn configure(
+        meta: &mut ConstraintSystem<F>,
+        tx_table: TxTable,
+        challenges: &Challenges<Expression<F>>,
+    ) -> Self {
+        let q_sign_start = meta.complex_selector();
+        let q_sign_step = meta.complex_selector();
+        let q_sign_end = meta.complex_selector();
+        let tag = meta.fixed_column();
+        let sign = meta.advice_column();
+        let sign_rlc_acc = meta.advice_column_in(SecondPhase);
+
+        let q_u128_start = meta.complex_selector();
+        let q_u128_step = meta.complex_selector();
+        let q_u128_end = meta.complex_selector();
+        let sign_u128_acc = meta.advice_column();
+
+        let q_check = meta.complex_selector();
+        let mul_add = MulAddChip::configure(meta, |meta| meta.query_selector(q_check));
+
+        let gx1_rlc = crate::evm_circuit::util::rlc::expr(
+            GX1.to_be_bytes()
+                .map(|v| Expression::Constant(F::from(v as u64)))
+                .as_ref(),
+            challenges.evm_word(),
+        );
+
+        let gx2_rlc = crate::evm_circuit::util::rlc::expr(
+            GX2.to_be_bytes()
+                .map(|v| Expression::Constant(F::from(v as u64)))
+                .as_ref(),
+            challenges.evm_word(),
+        );
+        let is_equal_gx2 = IsEqualChip::configure(
+            meta,
+            |meta| meta.query_selector(q_check),
+            |meta| meta.query_advice(sign_rlc_acc, Rotation(64)), // SigR == GX2
+            |_| gx2_rlc.expr(),
+        );
+
+        // signature rlc
+        meta.create_gate(
+            "sign_rlc_acc[i+1] = sign_rlc_acc[i] * randomness + sign[i+1]",
+            |meta| {
+                let mut cb = BaseConstraintBuilder::new(MAX_DEGREE);
+
+                let q_sign_step = meta.query_selector(q_sign_step);
+                let sign_rlc_acc_next = meta.query_advice(sign_rlc_acc, Rotation::next());
+                let sign_rlc_acc = meta.query_advice(sign_rlc_acc, Rotation::cur());
+                let sign = meta.query_advice(sign, Rotation::next());
+                let randomness = challenges.evm_word();
+                cb.require_equal(
+                    "sign_rlc_acc[i+1] = sign_rlc_acc[i] * randomness + sign[i+1]",
+                    sign_rlc_acc_next,
+                    sign_rlc_acc * randomness + sign,
+                );
+                cb.gate(q_sign_step)
+            },
+        );
+        meta.create_gate("sign_rlc_acc[0] = sign[0]", |meta| {
+            let mut cb = BaseConstraintBuilder::new(MAX_DEGREE);
+
+            let q_sign_start = meta.query_selector(q_sign_start);
+            let sign_rlc_acc = meta.query_advice(sign_rlc_acc, Rotation::cur());
+            let sign = meta.query_advice(sign, Rotation::cur());
+
+            cb.require_equal("sign_rlc_acc[0] = sign[0]", sign_rlc_acc, sign);
+            cb.gate(q_sign_start)
+        });
+        meta.lookup_any("sign_r or msg_hash in tx_table", |meta| {
```

Comment:
Could potentially use a copy constraint instead of a lookup right? I think also copy constraint could be used to replace PiTable? A bit trickier then just using lookups to share data so totally fine with the current approach, just checking if it would be possible.

Reply:
`AnchorCircuit` doesn't assign the `TxTable` and `Pitable` in `SuperCircuit`.  So I can't get the assigned cells from these two tables. But I found a trick approach to satisfy the purpose, just like [scroll](https://github.com/scroll-tech/zkevm-circuits/blob/develop/zkevm-circuits/src/pi_circuit.rs#L903):
```rust
use halo2_proofs::circuit::{Cell, RegionIndex};
region.constrain_equal(
    tx_hash_cell.cell(),
    Cell {
        region_index: RegionIndex(1), // FIXME: this is not safe
        row_offset: i * TX_LEN + TX_HASH_OFFSET,
        column: self.tx_table.value.into(),
    },
)?;
```
Hack the halo2's API, gives us the ability to create the cells